ShopLens/VoicedText/TextVoicers/TextVoicerSpeechSynthesizer.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System.ComponentModel;
+using System.Speech.Synthesis;
+
+namespace VoicedText.TextVoicers
+{
+    public class TextVoicerSpeechSynthesizer : ITextVoicer
+    {
+        private SpeechSynthesizer textVoicer;
+        private PromptBuilder messageBuilder;
+        private PromptStyle voicerSpeed;
+
+        /// <summary>
+        /// The possible speeds of the voicer.
+        /// </summary>
+        public enum VoicerSpeed { ExtraSlow, Slow, Normal, Fast, ExtraFast};
+
+        /// <summary>
+        /// The current speed of the voicer.
+        /// </summary>
+        public VoicerSpeed speedOfVoicer;
+
+        /// <summary>
+        /// The possible volumes of the voicer.
+        /// </summary>
+        public enum VoicerVolume { NoVolume, Low, Medium, High, Max};
+
+        /// <summary>
+        /// The current volume of the voicer.
+        /// </summary>
+        public VoicerVolume volumeOfVoicer;
+
+        /// <summary>
+        /// The class constructor.
+        /// </summary>
+        /// <remarks>
+        /// By default it sets the speed of the voicer to Normal
+        /// and the maximum volume of the voicer to Max.
+        /// </remarks>
+        public TextVoicerSpeechSynthesizer()
+        {
+            textVoicer = new SpeechSynthesizer();
+            messageBuilder = new PromptBuilder();
+
+            speedOfVoicer = VoicerSpeed.Normal;
+            volumeOfVoicer = VoicerVolume.Max;
+
+            SetSpeed((int)speedOfVoicer);
+            SetVolume((int)volumeOfVoicer);
+        }
+
+        /// <summary>
+        /// Sets the volume of the voicer.
+        /// </summary>
+        /// <param name="volumeOfVoicer">
+        /// Pass 0 for NoVolume, 1 for Low, 2 for Medium, 3 for High and 4 for Max volume level.
+        /// </param>
+        /// <returns></returns>
+        public void SetVolume(int volumeOfVoicer)
+        {
+            this.volumeOfVoicer = (VoicerVolume)volumeOfVoicer;
+
+            switch (volumeOfVoicer)
+            {
+                case (int)VoicerVolume.NoVolume:
+                    textVoicer.Volume = 0;
```

Comment:
Should probably be constants/enum, but will probably have to wait for @mrjslau 's AppConfig commits